Status: unsubmitted

This patch was for Debian bug #239535.  It needs to be tested, and
submitted.  It has shrunk since the first version, and may no
longer be necessary.

---
 gdb/valops.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: gdb-7.4/gdb/valops.c
===================================================================
--- gdb-7.4.orig/gdb/valops.c	2011-11-17 15:58:35.000000000 +0000
+++ gdb-7.4/gdb/valops.c	2011-11-17 17:00:55.453359139 +0000
@@ -3171,8 +3171,10 @@
 	return 1;
     }
 
+  /* Check each baseclass.  Call check_typedef, which will follow typedefs
+     and do opaque/stub type resolution.  */
   for (i = TYPE_N_BASECLASSES (type) - 1; i >= 0; i--)
-    if (check_field (TYPE_BASECLASS (type, i), name))
+    if (check_field (check_typedef (TYPE_BASECLASS (type, i)), name))
       return 1;
 
   return 0;
